File: be/src/models/SupportThread.js
const mongoose = require('mongoose');

const supportThreadSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
      index: true,
    },
    assignedAdminId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      default: null,
      index: true,
    },
    subject: {
      type: String,
      default: 'Support request',
      trim: true,
      maxlength: 120,
    },
    category: {
      type: String,
      enum: ['GENERAL', 'SAFETY', 'ACCOUNT', 'TECHNICAL', 'REPORT'],
      default: 'GENERAL',
      index: true,
    },
    status: {
      type: String,
      enum: ['OPEN', 'PENDING_USER', 'RESOLVED'],
      default: 'OPEN',
      index: true,
    },
    lastMessageAt: {
      type: Date,
      default: Date.now,
      index: true,
    },
  },
  { timestamps: true },
);

supportThreadSchema.index({ userId: 1, status: 1, lastMessageAt: -1 });

module.exports = mongoose.model('SupportThread', supportThreadSchema);

File: fe/lib/core/models/support_thread.dart
class SupportThread {
  const SupportThread({
    required this.id,
    required this.userId,
    required this.subject,
    required this.category,
    required this.status,
    required this.lastMessageAt,
  });

  final String id;
  final String userId;
  final String subject;
  final String category;
  final String status;
  final DateTime? lastMessageAt;

  factory SupportThread.fromJson(Map<String, dynamic> json) {
    return SupportThread(
      id: (json['id'] ?? json['_id'] ?? '').toString(),
      userId: (json['userId'] ?? '').toString(),
      subject: (json['subject'] ?? 'Support request').toString(),
      category: (json['category'] ?? 'GENERAL').toString(),
      status: (json['status'] ?? 'OPEN').toString(),
      lastMessageAt: json['lastMessageAt'] != null
          ? DateTime.tryParse(json['lastMessageAt'].toString())
          : null,
    );
  }
}
